pkg/flowkit/util/utilities.go

```diff
@@ -171,18 +171,17 @@ func AddToGitIgnore(filename string, loader ReaderWriter) error {
 	gitIgnorePath := path.Join(currentWd, ".gitignore")
 	gitIgnoreFiles := ""
 
-	fileStat, err := os.Stat(gitIgnorePath)
+	_, err = os.Stat(gitIgnorePath)
 	if !os.IsNotExist(err) { // if gitignore exists
 		gitIgnoreFilesRaw, err := loader.ReadFile(gitIgnorePath)
 		if err != nil {
 			return err
 		}
 		gitIgnoreFiles = string(gitIgnoreFilesRaw)
 	}
-
 	return loader.WriteFile(
 		gitIgnorePath,
 		[]byte(fmt.Sprintf("%s\n%s", gitIgnoreFiles, filename)),
-		fileStat.Mode().Perm(),
+		0644,
```

Comment:
we can still set the permission file has if it exists, have a default and assign if exists to the existing one.